### PR TITLE
Merge o1js-main one last time

### PR DESCRIPTION
**Pull request by mitschabaude**
After merging this, the plan is that o1js uses `develop`

### CODE REVIEW

File: src/lib/pickles/wrap_main.ml
Comment (mitschabaude) [inline src/lib/pickles/wrap_main.ml:1]: it's unfortunate how hard it is here to see actual changes. however, it does seem to me like all the additional `assert_consistent` checks ("Check consistency between index and feature flags") are not in `develop`, and I wonder why, given that this comes from berkeley?

**Comment by mitschabaude** [comment]
!ci-toolchain-me

**Review by MartinMinkov — APPROVED** [review]

**Review by dkijania — APPROVED** [review]

File: src/lib/pickles/wrap_main.ml [hunk 1]
```diff
@@ -129,319 +132,412 @@ let wrap_main
...
+                  Boolean.true_
+              | Opt.Maybe (b, _) ->
+                  b
+              | Opt.Nothing ->
+                  Boolean.false_
+            in
+            let assert_consistent comm flag =
+              Boolean.Assert.( = ) (commitment_flag comm) (Lazy.force flag)
+            in
+            assert_consistent xor_comm xor ;
```
Comment (dannywillems) [inline src/lib/pickles/wrap_main.ml:299]: Seems related to https://github.com/MinaProtocol/mina/pull/15234

**Review by dannywillems — APPROVED** [review]
Reviewed crypto subdirectory + pickles.

File: buildkite/src/Command/TestExecutive.dhall [hunk 2]
```diff
@@ -27,7 +27,13 @@ in
...
         key = "integration-test-${testName}",
         target = Size.Integration,
-        depends_on = dependsOn
+        depends_on = dependsOn,
+        retries = [
+          -- common/flake error
+          Command.Retry::{ exit_status = Command.ExitStatus.Code +1, limit = Some 4 },
+          -- Blindly retry 4 more times anyway. Why not.
+          Command.Retry::{ exit_status = Command.ExitStatus.Any, limit = Some 4 }
+        ]
```
[the remark this answers is not in the document]
Comment (mrmr1993) [reply buildkite/src/Command/TestExecutive.dhall:36]: Can you revert to the berkeley version? We don't want to do blind retries any more.

**Comment by mrmr1993** [comment]
!ci-nightly-me

File: src/app/test_executive/hard_fork.ml [hunk 3]
```diff
@@ -325,8 +325,9 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         ; authorization_kind = Signature
         }
       in
-      Transaction_snark.For_tests.deploy_snapp ~constraint_constants
-        zkapp_command_spec
+      Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/hard_fork.ml:328]: This should bind inside the monad instead of using a block.

File: src/app/heap_usage/values.ml [hunk 4]
```diff
@@ -36,8 +36,9 @@ let account : Mina_base.Account.t =
 let zkapp_command =
   let num_updates = 16 in
   let _ledger, zkapp_commands =
-    Snark_profiler_lib.create_ledger_and_zkapps ~min_num_updates:num_updates
-      ~num_proof_updates:num_updates ~max_num_updates:num_updates ()
+    Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/heap_usage/values.ml:39]: This should probably be a `_ Async.Deferred.t Lazy.t` type that we force and bind over.

File: flake.nix [hunk 5]
```diff
@@ -302,7 +302,9 @@
         # Main user-facing binaries.
         packages = rec {
           inherit (ocamlPackages)
-            mina devnet mainnet mina_tests mina-ocaml-format mina_client_sdk test_executive with-instrumentation;
+            mina devnet mainnet mina_tests mina-ocaml-format test_executive;
+          devnet-caqti-patched = pkgs.ocamlPackages_mina_caqti_patched.devnet;
+          mainnet-caqti-patched = pkgs.ocamlPackages_mina_caqti_patched.mainnet;
```
[the remark this answers is not in the document]
Comment (mrmr1993) [reply flake.nix:307]: These were deliberately deleted on develop. Please do the same here.

File: src/app/heap_usage/values.ml [hunk 6]
```diff
@@ -65,6 +66,7 @@ let verification_key =
     Transaction_snark.For_tests.create_trivial_snapp
       ~constraint_constants:Genesis_constants.Constraint_constants.compiled ()
   in
+  let vk = Async.Thread_safe.block_on_async_exn (fun () -> vk) in
```
Comment (mrmr1993) [inline src/app/heap_usage/values.ml:69]: As above, so below

File: src/app/test_executive/peers_reliability_test.ml [hunk 7]
```diff
@@ -139,9 +139,10 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
              }
            in
            return
-           @@ Transaction_snark.For_tests.deploy_snapp
-                ~constraint_constants:(Network.constraint_constants network)
-                parties_spec
+           @@ Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/peers_reliability_test.ml:142]: Monad bind

File: src/app/test_executive/verification_key_update.ml [hunk 8]
```diff
@@ -216,7 +220,8 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         ; authorization_kind = Signature
         }
       in
-      Transaction_snark.For_tests.deploy_snapp ~constraint_constants spec
+      Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/verification_key_update.ml:223]: Expose the `Deferred.t`, bind below.

File: src/app/test_executive/zkapps.ml [hunk 9]
```diff
@@ -179,8 +179,9 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         }
       in
       return
-      @@ Transaction_snark.For_tests.deploy_snapp ~constraint_constants
-           zkapp_command_spec
+      @@ Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/zkapps.ml:182]: Monad bind.

File: src/app/test_executive/zkapps_timing.ml [hunk 10]
```diff
@@ -105,8 +105,9 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
           Token_id.default
       in
 
-      ( Transaction_snark.For_tests.deploy_snapp ~constraint_constants
-          zkapp_command_spec
+      ( Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/zkapps_timing.ml:108]: bind

File: src/app/test_executive/zkapps_timing.ml [hunk 11]
```diff
@@ -150,8 +151,9 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         }
       in
       return
-      @@ Transaction_snark.For_tests.deploy_snapp ~constraint_constants
-           zkapp_command_spec
+      @@ Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/zkapps_timing.ml:154]: bind

File: src/app/test_executive/zkapps_timing.ml [hunk 12]
```diff
@@ -200,8 +202,9 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
           (zkapp_keypair.public_key |> Signature_lib.Public_key.compress)
           Token_id.default
       in
-      ( Transaction_snark.For_tests.deploy_snapp ~constraint_constants
-          zkapp_command_spec
+      ( Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/zkapps_timing.ml:205]: bind

File: src/app/test_executive/zkapps_timing.ml [hunk 13]
```diff
@@ -242,8 +245,9 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         ; authorization_kind = Signature
         }
       in
-      Transaction_snark.For_tests.deploy_snapp ~constraint_constants
-        zkapp_command_spec
+      Async.Thread_safe.block_on_async_exn (fun () ->
```
Comment (mrmr1993) [inline src/app/test_executive/zkapps_timing.ml:248]: bind

File: src/lib/mina_base/receipt.ml [hunk 14]
```diff
@@ -47,7 +47,10 @@ module Chain_hash = struct
 
   let equal = Stable.Latest.equal
 
-  let empty = of_hash Random_oracle.Legacy.(salt "CodaReceiptEmpty" |> digest)
+  let empty =
```
Comment (mrmr1993) [inline src/lib/mina_base/receipt.ml:50]: Revert.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-nightly-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-nightly-me

**Comment by mrmr1993** [comment]
Nightly run: https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/1937

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-nightly-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-nightly-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-nightly-me

**Pull request merged**